.github/workflows/release.yml

```diff
@@ -128,24 +137,35 @@ jobs:
       test_run: ${{ inputs.test_run }}
 
   pypi-release:
-    if: ${{ !inputs.test_run }}
+    name: PyPI Release
 
-    name: Pypi Release
+    needs: [github-release]
 
-    runs-on: ubuntu-latest
+    uses: dbt-labs/dbt-release/.github/workflows/pypi-release.yml@dev
 
-    needs: github-release
+    with:
+      version_number: ${{ inputs.version_number }}
+      test_run: ${{ inputs.test_run }}
 
-    environment: PypiProd
+    secrets:
+      PYPI_API_TOKEN: ${{ secrets.PYPI_API_TOKEN }}
+      TEST_PYPI_API_TOKEN: ${{ secrets.TEST_PYPI_API_TOKEN }}
+
+  slack-notification:
+    name: Slack Notification
+    if: ${{ failure() }}
+
+    needs:
+      [
+        bump-version-generate-changelog,
+        build-test-package,
+        github-release,
+        pypi-release,
+      ]
```

Comment:
This job will run if others are skipped since we use the conditional expression ([Related doc](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idneeds)).
Example: https://github.com/dbt-labs/dbt-spark-release-test/actions/runs/3873612452/jobs/6603839237